Kitodo/src/main/java/org/kitodo/production/services/data/base/SearchDatabaseService.java

```diff
@@ -226,4 +355,196 @@ protected String getSort(String sortField, SortOrder sortOrder) {
             return "";
         }
     }
+
+    /**
+     * Enters a search query into the log when it is running in debug level.
+     * Placeholders are replaced with their parameter values.
+     * 
+     * @param query
+     *            search query
+     * @param parameters
+     *            parameter values
+     */
+    private static void debugLogQuery(String query, Map<String, Object> parameters) {
+        debugLogQuery(query, parameters, Integer.MIN_VALUE, Integer.MIN_VALUE);
+    }
+
+    /**
+     * Enters a search query into the log when it is running in debug level.
+     * Placeholders are replaced with their parameter values.
+     * 
+     * @param query
+     *            search query
+     * @param parameters
+     *            parameter values
+     * @param initPointer
+     *            can initialize the object pointer to a later object (sets
+     *            {@linkplain Query#setFirstResult(int)})
+     * @param stopCount
+     *            the search stops after count hits (sets
+     *            {@linkplain Query#setMaxResults(int)})
+     */
+    private static void debugLogQuery(String query, Map<String, Object> parameters, int initPointer, int stopCount) {
+        if (logger.isDebugEnabled()) {
+            String resolved = PARAMETER_PATTERN.matcher(query).replaceAll(matchResult -> {
+                Object parameter = parameters.get(matchResult.group(1));
+                if (Objects.isNull(parameter)) {
+                    return matchResult.group();
+                }
+                if (parameter instanceof String) {
+                    return '\'' + ((String) parameter) + '\'';
+                }
+                if (parameter instanceof AbstractCollection) {
+                    return Objects.toString(parameter).replaceFirst("^\\[(.*)\\]$", "$1");
+                }
+                return Objects.toString(parameter);
+            });
+            if (initPointer != Integer.MIN_VALUE || stopCount != Integer.MIN_VALUE) {
+                if (stopCount != Integer.MIN_VALUE) {
+                    resolved = String.format("%s (limit=%d)", resolved, stopCount);
+                } else {
+                    resolved = String.format("%s (limit=%d, offset=%d)", resolved, stopCount, initPointer);
+                }
+            }
+            logger.debug(resolved);
+        }
+    }
+
+    // === functions no longer used ===
```

Comment:
Why you are adding this methods below this line and marking them as  deprecated if they are not in use anymore? Why not delete them?

Reply:
I took your feedback as an opportunity to take another look at it yesterday, and was able to remove quite a bit of dead code.